src/components/heading/heading.test.tsx

```diff
@@ -0,0 +1,239 @@
+  expect(helpIconLink).toHaveAttribute(
+    "href",
+    "https://www.warnerbros.com/movies/heat"
+  );
+});
+
+test("renders wih an 'aria-label' attribute on the help, via the `aria-label` prop", () => {
+  render(
+    <Heading
+      title="foo"
+      help="help text"
+      helpAriaLabel="help-aria-label"
+      helpLink="https://www.warnerbros.com/movies/heat"
+    />
+  );
+
+  const helpIconLink = screen.getByRole("link", { name: "help-aria-label" });
+  expect(helpIconLink).toBeVisible();
+});
+
+test("renders a back link, when `backLink` prop is a href", () => {
+  render(
+    <Heading title="foo" backLink="https://www.warnerbros.com/movies/heat" />
+  );
+
+  const backLink = screen.getByRole("link", { name: "Back" });
+  expect(backLink).toHaveAttribute(
+    "href",
+    "https://www.warnerbros.com/movies/heat"
+  );
+});
+
+test("focuses the back link, when `backLink` is a href on mouse down", () => {
+  render(
+    <Heading title="foo" backLink="https://www.warnerbros.com/movies/heat" />
+  );
+
+  const backLink = screen.getByRole("link", { name: "Back" });
+  fireEvent.mouseDown(backLink);
+
+  expect(backLink).toBeFocused();
+});
+
+test("renders a back button, when `backLink` prop is a function", () => {
+  const backLinkFunction = jest.fn();
+  render(<Heading title="foo" backLink={backLinkFunction} />);
+
+  const backLink = screen.getByRole("button", { name: "Back" });
+  expect(backLink).toBeVisible();
+});
+
+test("focuses the back link, when `backLink` prop is a function", () => {
+  const backLinkFunction = jest.fn();
+  render(<Heading title="foo" backLink={backLinkFunction} />);
+
+  const backLink = screen.getByRole("button", { name: "Back" });
+  fireEvent.mouseDown(backLink);
+
+  expect(backLink).toBeFocused();
+});
+
+/* styling test for coverage */
+test("alters the grid-column property on the subtitle node, when a back link/button is rendered", () => {
+  render(
+    <Heading
+      title="foo"
+      subheader="bar"
+      backLink="https://www.warnerbros.com/movies/heat"
+    />
+  );
+
+  const subtitle = screen.getByTestId("subtitle");
+
+  expect(subtitle).toHaveStyle("grid-column: 2");
+});
+
+/* styling test for coverage */
+test("alters the outline property on the back link/button, when the back link is focused and consumers opt out of the focus redesign", () => {
+  render(
+    <CarbonProvider focusRedesignOptOut>
+      <Heading title="foo" backLink="https://www.warnerbros.com/movies/heat" />
+    </CarbonProvider>
+  );
+
+  const backLink = screen.getByTestId("heading-back-button");
+  backLink.focus();
+
+  expect(backLink).toHaveStyleRule(
+    "outline",
+    "3px solid var(--colorsSemanticFocus500)",
+    { modifier: "button:focus" }
+  );
+});
+
+test("renders a divider when the `divider` prop is true", () => {
+  render(<Heading title="foo" divider />);
+
+  const separator = screen.getByRole("separator");
+  expect(separator).toBeVisible();
```

Comment:
**comment**: while this test is better than the following one for the separator (I'd recommend reading my long comment on that first for context), I'm still not really a fan, since this test will pass even if we accidentally stop the divider being rendered, if there's any `<hr>` being rendered elsewhere (including if we accidentally always render the separator).

This isn't as easy to fix because the unlike the `StyledDivider` doesn't have a `data-role`. I don't see why we couldn't add one though, to make this test more robust.